cloud/disk_manager/test/acceptance/test_runner/cleanup.py

```diff
@@ -0,0 +1,124 @@
+        [lister, deleter] = handlers
+        for entity in lister():
+            ttl = entity_ttls.get(entity_type, None)
+            if ttl is None:
+                _logger.warning("No ttl for entity %s", entity_type)
+                continue
+            should_be_older_than = datetime.now() - ttl
+            if entity.created_at > should_be_older_than:
+                continue
+            for pattern in regexps.get(entity_type, []):
+                if re.match(pattern, entity.name):
+                    _logger.info(
+                        "Deleting %s with name %s",
+                        entity_type, entity.name)
+                    try:
+                        deleter(entity)
+                    except Exception as e:
+                        _logger.error(
+                            "Error while deleting %s with name %s",
+                            entity_type,
+                            entity.name,
+                            exc_info=e,
+                        )
+                    break
+
+
+def cleanup_previous_acceptance_tests_results(
+        ycp: YcpWrapper,
+        test_type: str,
+        disk_size: int,
+        disk_blocksize: int,
+        entity_ttl: timedelta = timedelta(days=1),
+):
+    entity_accessors = _get_entity_accessors(ycp)
+    disk_size = size_prettifier(disk_size * (1024 ** 3)).lower()
+    disk_blocksize = size_prettifier(disk_blocksize).lower()
+    _logger.info(
+        "Performing cleanup for %s disk size %s, disk block size %s",
+        test_type,
+        disk_size,
+        disk_blocksize,
+    )
+    regexps = {
+        entity_type: [
+            re.compile(
+                fr'^acceptance-test-{entity_type}-'
+                fr'{test_type}-{disk_size}-{disk_blocksize}-[0-9]+$',
+            ),
+            re.compile(fr'^acceptance-test-{entity_type}-{test_type}-[0-9]+$')
+        ] for entity_type in entity_accessors
+    }
+    _cleanup_stale_entities(
+        entity_accessors,
+        {
+            'disk': entity_ttl,
+            'image': entity_ttl,
+            'snapshot': entity_ttl,
+        },
+        regexps,
+    )
+
+
+class BaseResourceCleaner:
+    def __init__(self, ycp: YcpWrapper, args: argparse.Namespace):
+        self._ycp = ycp
+        self._args = args
+        self._entity_ttls: dict[str, timedelta] = {
+            'instance': timedelta(days=1),
+            'disk': timedelta(days=1),
+            'snapshot': timedelta(days=1),
+            'images': timedelta(days=1)
+        }
+        self._patterns: dict[str, list[re.Pattern]] = {
+            'instance':  [
+                re.compile(rf'^acceptance-test-{args.test_type}-[0-9]+$')
+            ],
+        }
+
+    def cleanup(self):
+        accessors = {
+            **_get_entity_accessors(self._ycp),
+            'instance': (
+                self._ycp.list_instances,
+                self._ycp.delete_instance,
+            )
+        }
+        accessors.pop('image')
```

Comment:
What does it mean?

Reply:
We do not need image accessor for this kind of cleanup. So it is removed. We can simplfy this by relaxing condition in the base cleanup function.

Reply:
What would happen if 'image accessor' remains here?

Reply:
Simplified entity accessors dict construction.  Accessor's dict shall have the same keys as ttl's dict. It would be a smart idea to put them into a class, but it would go out of the scope of the current review.